Card/src/main/java/com/paypal/android/card/CardClient.kt

```diff
@@ -1,42 +1,155 @@
 package com.paypal.android.card
 
+import android.net.Uri
+import androidx.fragment.app.FragmentActivity
+import com.braintreepayments.api.BrowserSwitchClient
+import com.braintreepayments.api.BrowserSwitchOptions
+import com.braintreepayments.api.BrowserSwitchResult
+import com.braintreepayments.api.BrowserSwitchStatus
+import com.paypal.android.card.api.CardAPI
+import com.paypal.android.card.api.ConfirmPaymentSourceResponse
+import com.paypal.android.card.threedsecure.ThreeDSecureRequest
+import com.paypal.android.card.threedsecure.ThreeDSecureResult
 import com.paypal.android.core.API
 import com.paypal.android.core.CoreConfig
 import com.paypal.android.core.PayPalSDKError
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
 
 /**
  * Use this client to approve an order with a [Card].
  */
-class CardClient internal constructor(private val cardAPI: CardAPI) {
+class CardClient internal constructor(
+    private val activity: FragmentActivity,
+    private val cardAPI: CardAPI,
+    private val coroutineContext: CoroutineContext // this could be moved to each approve function
+) {
 
-    constructor(configuration: CoreConfig) :
-            this(CardAPI(API(configuration)))
+    private val browserSwitchClient = BrowserSwitchClient()
+    private var approveOrderCallback: ApproveOrderCallback? = null
+    private var confirmPaymentSourceResponse: ConfirmPaymentSourceResponse? = null
+    private var lifeCycleObserver: CardLifeCycleObserver? = null
 
     /**
-     * Confirm [Card] payment source for an order. Use this method for Kotlin integrations
+     *  CardClient constructor
      *
-     * @param request [CardRequest] for requesting an order approval
+     *  @param [activity] Activity that launches the card client
+     *  @param [configuration] Configuration parameters for client
+     *  @param [coroutineContext] CoroutineContext to execute
      */
-    suspend fun approveOrder(request: CardRequest): CardResult =
```

Comment:
qq: We're removing `suspend` methods?

Reply:
I thought it was the only way, because 3DS does a browser switch, so we cant just return a value.

Reply:
@sshropshire i did add a `CoroutineContext` to the client, i think we should give developers the option to choose in which thread our client is going to do its request. Right now its in the `constructor`, but maybe it should be a param for each function

Reply:
> I thought it was the only way, because 3DS does a browser switch, so we cant just return a value.

Ok yeah that makes sense. Then we should remove the callback parameter too and replace it with a listener.